internal/pkg/automation/fuzzer.go

```diff
@@ -0,0 +1,168 @@
+package automation
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/lawrencegripper/azbrowse/internal/pkg/config"
+	"github.com/lawrencegripper/azbrowse/internal/pkg/eventing"
+	"github.com/lawrencegripper/azbrowse/internal/pkg/expanders"
+	"github.com/lawrencegripper/azbrowse/internal/pkg/views"
+	"github.com/stuartleeks/gocui"
+)
+
+// ErrorReporter Tracks errors during fuzzing
+type ErrorReporter struct {
+}
+
+// Errorf adds an error found while fuzzing
+func (e *ErrorReporter) Errorf(format string, args ...interface{}) {
+	fmt.Printf(format, args...)
+	os.Exit(5)
+}
+
+// StartAutomatedFuzzer will start walking the tree of nodes
+// logging out information and running tests while walking the tree
+func StartAutomatedFuzzer(list *views.ListWidget, settings *config.Settings, gui *gocui.Gui) {
+
+	type fuzzState struct {
+		current int
+		max     int
+	}
+
+	min := func(a, b int) int {
+		if a < b {
+			return a
+		}
+		return b
+	}
+
+	testFunc := func(currentNode *expanders.TreeNode, nodes []*expanders.TreeNode) {
+		// if r := regexp.MustCompile(".*/providers/Microsoft.ContainerRegistry/registries/.*/"); r.MatchString(currentNode.ID) {
+		// 	st.Expect(&ErrorReporter{}, len(nodes), 5)
+		// }
+	}
+
+	shouldSkip := func(currentNode *expanders.TreeNode, itemId string) (shouldSkip bool, maxToExpand int) {
```

Comment:
There's currently no circumstances under which shouldSkip is returned true - is this just for future proofing in case you want to add other conditions?

Reply:
Yeah for some more universal providers like `Activity Log` I only really want to ever walk it once no matter how many resources have it. Added this now with a `defer` statement... or attempting to.